src/model/Stroke.cpp

```diff
@@ -359,42 +395,32 @@ void Stroke::calcSize() const {
         Element::snappedBounds = Rectangle<double>{};
     }
 
-    double minX = DBL_MAX;
-    double maxX = DBL_MIN;
-    double minY = DBL_MAX;
-    double maxY = DBL_MIN;
-
     double minSnapX = DBL_MAX;
     double maxSnapX = DBL_MIN;
     double minSnapY = DBL_MAX;
     double maxSnapY = DBL_MIN;
 
-    bool hasPressure = points[0].z != Point::NO_PRESSURE;
-    double halfThick = this->width / 2.0;  //  accommodate for pen width
+    auto halfThick = 0.0;
 
+    //#pragma omp parralel
     for (auto&& p: points) {
-        if (hasPressure) {
-            halfThick = p.z / 2.0;
-        }
-
-        minX = std::min(minX, p.x - halfThick);
-        minY = std::min(minY, p.y - halfThick);
-
-        maxX = std::max(maxX, p.x + halfThick);
-        maxY = std::max(maxY, p.y + halfThick);
-
         minSnapX = std::min(minSnapX, p.x);
         minSnapY = std::min(minSnapY, p.y);
-
         maxSnapX = std::max(maxSnapX, p.x);
         maxSnapY = std::max(maxSnapY, p.y);
     }
 
+    halfThick = points[0].z != Point::NO_PRESSURE ? halfThick / 2.0 : this->width / 2.0;
```

Comment:
This looks flawed to me. `halfThick` was set to `0.0`, so `halfThick / 2.0` will simply be `0.0`. Was your intention to take `halfThick` to be the maximal `p.z` divided by 2?
This will potentially enlarge the `bounds` rectangle (compared to the current implementation) when pressure is non-constant.

Also do you assume that either all points of a stroke have no pressure or all points have pressure?

Reply:
> Was your intention to take halfThick to be the maximal p.z divided by 2
Yes, thanks for the catch.

> This will potentially enlarge the bounds rectangle (compared to the current implementation) when pressure is non-constant.
I am aware of that, but I think this is not such a big problem;
 - The thickness will have a small range
 - The previous attempt also was an approximation.
 - The rectangle is mostly used to shorten hard computations

Reply:
> Also do you assume that either all points of a stroke have no pressure or all points have pressure?

This hasn't changed, so that assumption was made before.

Reply:
Merging tomorrow, if no objections.

Reply:
> Also do you assume that either all points of a stroke have no pressure or all points have pressure?

This assumption should be true currently. The only case where no pressure is available is if you use a tool that has no pressure sensitivity or where GDK+ does not supply it. If GDK+ does not supply it, this is as far as I have seen, true for all events. Though there is nothing that ensures this, so if this should ever change, it will be a hard-to-find bug.

Reply:
[This line](https://github.com/xournalpp/xournalpp/blob/97c20f62c5a5c16ce6dbb21ffd19240f4fe55961/src/control/tools/StrokeHandler.cpp#L121) ensures that the pressure is never set if the first event"s pressure is not available.